packages/xo-web/src/xo-app/sr/tab-general.js

```diff
@@ -1,25 +1,161 @@
 import _ from 'intl'
 import Component from 'base-component'
+import decorate from 'apply-decorators'
 import HomeTags from 'home-tags'
 import Icon from 'icon'
-import map from 'lodash/map'
 import React from 'react'
 import Usage, { UsageElement } from 'usage'
 import { addTag, removeTag, getLicense } from 'xo'
 import { connectStore, formatSize } from 'utils'
 import { Container, Row, Col } from 'grid'
-import { createGetObject } from 'selectors'
-import { renderXoItemFromId } from 'render-xo-item'
+import {
+  createCollectionWrapper,
+  createGetObjectsOfType,
+  createSelector,
+} from 'selectors'
+import {
+  find,
+  flatMap,
+  forEach,
+  groupBy,
+  keyBy,
+  map,
+  mapValues,
+  sumBy,
+} from 'lodash'
+import { injectState, provideState } from 'reaclette'
 
-const UsageTooltip = connectStore(() => ({
-  vbd: createGetObject((_, { vdi }) => vdi.$VBDs[0]),
-}))(({ vbd, vdi }) => (
-  <span>
-    {vdi.name_label} − {formatSize(vdi.usage)}
-    {vbd != null && <br />}
-    {vbd != null && renderXoItemFromId(vbd.VM)}
-  </span>
-))
+const styles = { margin: '0.2em', padding: '0.1em' }
+const ulStyles = { margin: 0, padding: '0.1em' }
+
+const UsageTooltip = decorate([
+  connectStore(() => {
+    const getVbds = createGetObjectsOfType('VBD').pick(
+      createCollectionWrapper(
+        createSelector(
+          (_, { group }) => group.vdis,
+          vdis => flatMap(vdis, _ => _.$VBDs)
+        )
+      )
+    )
+
+    const getVms = createGetObjectsOfType('VM').pick(
+      createCollectionWrapper(
+        createSelector(
+          getVbds,
+          vbds => map(vbds, 'VM')
+        )
+      )
+    )
+
+    return {
+      vbds: getVbds,
+      vms: getVms,
+    }
+  }),
+  provideState({
+    computed: {
+      vdis: (_, { group: { vdis } }) => keyBy(vdis, 'id'),
+      vmsByVdi: ({ vdis }, { vbds, vms }) =>
+        mapValues(vdis, vdi => {
+          const vbd = find(vbds, { VDI: vdi.id })
+          return vbd && vms[vbd.VM]
+        }),
+    },
+  }),
+  injectState,
+  ({
+    state: { vmsByVdi },
+    group: { baseCopies, id, name_label: name, usage, vdis, snapshots },
+  }) => {
+    const showBaseCopies = baseCopies !== undefined && baseCopies.n > 0
+    const showVdis = showBaseCopies || snapshots.length > 0
+    let disk
+    let vm
+    return (
+      <div>
+        {vdis !== undefined ? (
+          <ul style={ulStyles}>
+            {showBaseCopies && (
+              <li>
+                {_('baseCopyTooltip', {
+                  n: baseCopies.n,
+                  usage: formatSize(baseCopies.usage),
+                })}
+              </li>
+            )}
+            {snapshots.length > 0 && (
+              <li>
+                {_('snapshotsTooltip', {
+                  n: snapshots.length,
+                  usage: formatSize(sumBy(snapshots, 'usage')),
+                })}
+              </li>
+            )}
+            {snapshots.length > 0 && (
+              <ul style={styles}>
```

Comment:
I don't think a [`ul`](https://devdocs.io/html/element/ul) can be a direct child of another `ul`.

Reply:
It's not a direct child. The first child of `ul` is [`li`](https://github.com/vatesfr/xen-orchestra/pull/3830/files#diff-3c43ab115a44db3dec53dd8cd4708c57R87)

Reply:
```html
<ul>
  <li>
  <li>
  <ul>
    <li>
```
so the second `ul` *is* a direct child of the first `ul`.